src/utils/web/formUtils.ts

```diff
@@ -21,7 +21,21 @@ export type GenericErrorFormValues = {
 export async function triggerServerActionForForm<
   F extends UseFormReturn<any, any, any>,
   P,
-  Fn extends (args: P) => Promise<{ errors: Record<string, string[]> } | object | undefined>,
+  Fn extends (args: P) => Promise<
+    | {
+        errors: Record<string, string[]>
+        errorsMetadata?: {
+          field: string
+          data: {
+            triggerException: boolean
+            severityLevel?: SeverityLevel
+            message?: string
+          }
```

Comment:
My understand is that if `triggerException` is `false`, we won't have `severityLevel` and `message`, if that's the case we could change this type to reflect that better
```suggestion
          data: {
            triggerException: false
            severityLevel?: never
            message?: never
          } | {
            triggerException: true
            severityLevel: SeverityLevel
            message: string
          }
```

Doing it like this would make both fields required when you set `triggerException` as true

Reply:
thanks man, really appreciate it